grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -743,6 +743,68 @@ class ProductAvailabilityService {
         return availableBinLocations
     }
 
+    AvailableItem getAvailableItemWithDefaultLots(Location origin, String productCode) {
+        Product product = Product.findByProductCode(productCode)
+        List<AvailableItem> availableItems = getAvailableBinLocations(origin, product?.id)
+
+        List<AvailableItem> itemsWithDefaultLot = availableItems.findAll {
+            !it?.inventoryItem?.lotNumber // Null or empty lot number
+        }
+        // only infer if there is one possible value
+        if (itemsWithDefaultLot.size() == 1) {
+            return itemsWithDefaultLot.first()
+        }
+        return null
+    }
+
+    AvailableItem inferAvailableItemByBinLocation(Location origin, String productCode, String binLocationName) {
```

Comment:
I think `get` fits more than `infer` - we should change it to `get` here or use `infer` in all methods.